pilot/pkg/kube/inject/inject.go

```diff
@@ -540,6 +540,13 @@ func flippedContains(needle, haystack string) bool {
 func InjectionData(sidecarTemplate, valuesConfig, version string, deploymentMetadata *metav1.ObjectMeta, spec *corev1.PodSpec,
 	metadata *metav1.ObjectMeta, proxyConfig *meshconfig.ProxyConfig, meshConfig *meshconfig.MeshConfig) (
 	*SidecarInjectionSpec, string, error) {
+
+	// If DNSPolicy is not ClusterFirst, the envoy may not able to connect to pilot.
+	if spec.DNSPolicy != corev1.DNSClusterFirst {
+		log.Warnf("%q's DNSPolicy is not %s, the envoy may not able to connect to pilot\n",
```

Comment:
nit: namesapce/name

Reply:
Please capitalize "Envoy" and "Pilot" in the message  and comment.  Suggestion: `"%q's DNSPolicy is not %q.  The Envoy sidecar may not able to connect to Istio Pilot\n"`

Reply:
updated.